File: sdks/go/corthan/developer.go
package corthan

import (
	"context"
	"fmt"
	"net/http"
)

// DeveloperService wraps Developer API key operations.
type DeveloperService struct {
	t *TransportClient
}

// ListKeys retrieves masked developer API keys.
func (s *DeveloperService) ListKeys(ctx context.Context) (*DeveloperKeyListSuccess, error) {
	var result DeveloperKeyListSuccess
	err := s.t.Request(ctx, http.MethodGet, "/developers/keys", nil, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// CreateKey creates a new developer API key, returning the plaintext secret exactly once.
func (s *DeveloperService) CreateKey(ctx context.Context, name string) (*DeveloperKeyCreateSuccess, error) {
	body := map[string]string{
		"name": name,
	}
	var result DeveloperKeyCreateSuccess
	err := s.t.Request(ctx, http.MethodPost, "/developers/keys", body, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// RevokeKey revokes a developer API key by ID.
func (s *DeveloperService) RevokeKey(ctx context.Context, id string) error {
	path := fmt.Sprintf("/developers/keys/%s", id)
	return s.t.Request(ctx, http.MethodDelete, path, nil, nil)
}

File: sdks/node/src/services/developer.ts
import { TransportClient } from "../transport.js";
import { DeveloperKeyCreateSuccess, DeveloperKeyListSuccess } from "../types.js";

/**
 * Service for managing developer access keys.
 */
export class DeveloperService {
  constructor(private transport: TransportClient) {}

  /**
   * Generates a new API developer access credential key.
   */
  public async createKey(name: string): Promise<DeveloperKeyCreateSuccess> {
    return this.transport.request<DeveloperKeyCreateSuccess>("POST", "/developers/keys", {
      name
    });
  }

  /**
   * Lists active developer access credentials created for the tenant.
   */
  public async listKeys(): Promise<DeveloperKeyListSuccess> {
    return this.transport.request<DeveloperKeyListSuccess>("GET", "/developers/keys");
  }

  /**
   * Revokes a developer API access key by its identifier.
   */
  public async revokeKey(id: string): Promise<any> {
    return this.transport.request<any>("DELETE", `/developers/keys/${id}`);
  }
}
